provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -1381,13 +1431,20 @@ func nodeGroupProperties(cluster, v2 bool) map[string]schema.PropertySpec {
 				"mutually exclusive.",
 		},
 		"clusterIngressRule": {
-			TypeSpec:    schema.TypeSpec{Ref: awsRef("#/resources/aws:ec2%2FsecurityGroupRule:SecurityGroupRule")},
+			TypeSpec: schema.TypeSpec{
+				Ref:   awsRef("#/resources/aws:ec2%2FsecurityGroupRule:SecurityGroupRule"),
+				Plain: true,
+			},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/nodegroup.ts#L79-L82